src/master-playlist-controller.js

```diff
@@ -615,6 +619,9 @@ export class MasterPlaylistController extends videojs.EventTarget {
         // if we shouldn't switch to the next playlist, do nothing
         if (!this.shouldSwitchToMedia_(nextPlaylist)) {
           this.logger_(`earlyabort triggered, but we will not be switching from ${currentPlaylist.id} -> ${nextPlaylist.id}.`);
+          this.mainSegmentLoader_.earlyabortTimer = window.setTimeout(() => {
```

Comment:
We should clear this timer every time we come into this function right?

Reply:
Yeah, I have that as part of the TODOs. I wasn't sure whether we had a `player.setTimeout` method available here or whether I needed to make sure that the timer is handled fully as appropriate.
The variable should be renamed to include a `_` as well.